providers/flagd/pkg/configuration.go

```diff
@@ -0,0 +1,116 @@
+package flagd
+
+import (
+	"fmt"
+	"github.com/go-logr/logr"
+	"github.com/open-feature/go-sdk-contrib/providers/flagd/pkg/cache"
+	"os"
+	"strconv"
+)
+
+// Naming and defaults must comply with flagd environment variables
+const (
+	defaultMaxCacheSize                               int    = 1000
+	defaultPort                                              = 8013
+	defaultMaxEventStreamRetries                             = 5
+	defaultTLS                                        bool   = false
+	defaultCache                                      string = "lru"
+	defaultHost                                              = "localhost"
+	flagdHostEnvironmentVariableName                         = "FLAGD_HOST"
+	flagdPortEnvironmentVariableName                         = "FLAGD_PORT"
+	flagdTLSEnvironmentVariableName                          = "FLAGD_TLS"
+	flagdSocketPathEnvironmentVariableName                   = "FLAGD_SOCKET_PATH"
+	flagdServerCertPathEnvironmentVariableName               = "FLAGD_SERVER_CERT_PATH"
+	flagdCacheEnvironmentVariableName                        = "FLAGD_CACHE"
+	flagdMaxCacheSizeEnvironmentVariableName                 = "FLAGD_MAX_CACHE_SIZE"
+	flagdMaxEventStreamRetriesEnvironmentVariableName        = "FLAGD_MAX_EVENT_STREAM_RETRIES"
+	cacheDisabledValue                                       = cache.DisabledValue
+	cacheLRUValue                                            = cache.LRUValue
+	cacheInMemValue                                          = cache.InMemValue
+)
+
+type providerConfiguration struct {
+	CacheType                        string
+	CertificatePath                  string
+	EventStreamConnectionMaxAttempts int
+	Host                             string
+	MaxCacheSize                     int
+	OtelIntercept                    bool
+	Port                             uint16
+	SocketPath                       string
+	TLSEnabled                       bool
+}
+
+func newDefaultConfiguration() *providerConfiguration {
+	return &providerConfiguration{
+		CacheType:                        defaultCache,
+		EventStreamConnectionMaxAttempts: defaultMaxEventStreamRetries,
+		Host:                             defaultHost,
+		MaxCacheSize:                     defaultMaxCacheSize,
+		Port:                             defaultPort,
+		TLSEnabled:                       defaultTLS,
+	}
+}
+
+// updateFromEnvVar is a utility to update configurations based on current environment variables
+func (cfg *providerConfiguration) updateFromEnvVar(logger logr.Logger) {
```

Comment:
I see that the logger is not part of `providerConfiguration` but it feels weird to have it as an argument of a function since it could vary between calls to this function.

Reply:
Updated with the commit - https://github.com/open-feature/go-sdk-contrib/pull/317/commits/158f552c75623a69b753f67afb835e14b4ddbeaf

I made logger part of the config instance creation so that we don't have to call it with a logger